libraries/GCS_MAVLink/GCS_MAVLink.h

```diff
@@ -70,7 +70,12 @@ void comm_send_buffer(mavlink_channel_t chan, const uint8_t *buf, uint8_t len);
 uint16_t comm_get_txspace(mavlink_channel_t chan);
 
 #define MAVLINK_USE_CONVENIENCE_FUNCTIONS
+
+#pragma GCC diagnostic push
```

Comment:
What made this suddenly required?

Reply:
the strncpy() changes in this PR:
https://github.com/ArduPilot/pymavlink/pull/922
AFAICS the warning is bogus, the strncpy is valid, and yes, it is truncating a string, but that is what it is supposed to do
we could avoid by going back to memset/memcpy, could put the pragmas in pymavlink, but this was the simplest fix